idl/response/foxsi_get_source_map_spectrum.pro

```diff
@@ -0,0 +1,101 @@
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;FUNCTION:      "foxsi_get_source_map_spectrum"         
+;;;
+;;;HISTORY:       Initial Commit - 08/31/15 - Samuel Badman
+;;;                                                                                            
+;;;DESCRIPTION:   Funtion which returns an array of structures
+;;;               consisting of 2D maps with spectral information
+;;;               appended as tags. Default spectrum for the routine 
+;;;               get_foxsi_spectral_image.pro          
+;;;                                                                                            
+;;;CALL SEQUENCE: source_map_spectrum = foxsi_get_source_map_spectrum()                      
+;;;                                                                                            
+;;;KEYWORDS:      dx, dy - binsize of pixels in arcseconds 
+;;;               xc, yc - centre of image in solar coordinates (arcseconds)
+;;;                                                                                            
+;;;                                                                                            
+;;;COMMENTS:      Currently returns an array of maps with 2 Gaussian sources,
+;;;               each of FWHM 1/20 * FOV size and each with a peak count rate
+;;;               per bin of 1000. Source 1 is centred about [0.375,0.625], 
+;;;               Source 2 is centred about [0.875,0.625] expressed as a
+;;;               fraction of the FOV dimensions. In the spectral
+;;;               dimension, source 1 is constant in energy and source
+;;;               2 is attenuated  with a half width of ~20kev
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+FUNCTION foxsi_get_source_map_spectrum, dx = dx, dy = dy, xc = xc, yc = yc
+
+;;;;Define keyword defaults to 1 arcsecond per pixel and centre the image at the solar origin
+DEFAULT, dx, 1
+DEFAULT, dy, 1
+DEFAULT, xc, 0
+DEFAULT, yc, 0
+
+spectral_resolution = 2.0 ; keV per bin
+spectrum_lower_bound = 1.0
+spectrum_upper_bound = 60.0
+spectrum_dimension = (spectrum_upper_bound - spectrum_lower_bound)/spectral_resolution
+IF spectrum_upper_bound EQ spectrum_lower_bound THEN spectrum_dimension = 1
+
+source_array = DBLARR(100,100)
+;;;; Warning: changing the above dimensions of source_array will
+;;;; significantly affect the code runtime.
+
+;Make structure to act as a placeholder while making the source cube
+
+cube_creator = ADD_TAG(ADD_TAG(ADD_TAG(ADD_TAG(MAKE_MAP(source_array, dx=dx, dy=dy,    $
```

Comment:
To simplify, each map only needs the lower and upper value of the energy bin and nothing else. The 'global' tags (which apply to the array) such as spectrum_lower_bound are confusing. Once you have those values than spectral_resolution and energy are no longer necessary.